app/src/main/java/session/SharedPreferenceRepository.java

```diff
@@ -16,24 +16,25 @@
 import domain.Coupon;
 import domain.Offer;
 
+/**
+ * Used for saving coupons into memory on the mobile device. This will be replaced by a proper
+ * data storage system.
+ */
 public class SharedPreferenceRepository implements ISessionRepository {
     private final Gson gson;
-    private Context contextOfApplication;
     private SharedPreferences prefs;
 
-
     public SharedPreferenceRepository(Context contextOfApplication) {
-        this.contextOfApplication = contextOfApplication;
         this.prefs = PreferenceManager.getDefaultSharedPreferences(contextOfApplication);
         this.gson = new Gson();
     }
 
-    private List<Coupon> GetUserCouponsInPref(String userId){
+    private List<Coupon> getUserCouponsInPref(String userId){
         Type couponListType = new TypeToken<ArrayList<Coupon>>(){}.getType();
 
         String couponsJson = prefs.getString(userId, "");
 
-        if(couponsJson == null || couponsJson.isEmpty())
+        if(couponsJson.isEmpty())
             return null;
 
         List<Coupon> coupons  = gson.fromJson(couponsJson, couponListType);
```

Comment:
When running the code the coupons retrieved here does not have an offer attached. 
A null pointer exception is thrown when calling coupon.getOffer().getTitle() because offer is null.
Needs debugging